modules/internal/ChapelLocale.chpl

```diff
@@ -48,8 +48,32 @@ module ChapelLocale {
     // The parent of the root locale is nil (by definition).
     const parent : locale;
 
-    // To be removed from the required interface once legacy code is adjusted.
-    // Modified in RootLocale.init().
+    pragma "no doc" var nPUsPhysAcc: int;    // HW cores, OS accessible
+    pragma "no doc" var nPUsPhysAll: int;    // HW cores, all
+    pragma "no doc" var nPUsLogAcc: int;     // HW threads, OS accessible
+    pragma "no doc" var nPUsLogAll: int;     // HW threads, all
+
+    /*
+      A *processing unit* or *PU* is an instance of the processor
+      architecture, basically the thing that executes instructions.
+      ``numPUs()`` tells how many of these are present on this locale.
+      If ``physical==true`` (the default) this is the number of physical
+      PUs, commonly known as *cores*.  Otherwise it is the number of
+      hardware threads (such as hyperthreads), which on a multithreaded
+      hardware architecture would typically be more numerous.  If
+      ``accessible==true`` (the default) then the count reflects only
+      those PUs the program has access to, due to limiting by the OS for
```

Comment:
"limiting by" -> "limitations imposed by"? or "OS-imposed limitations"?  

That said, I'm not sure that, even with smoothing out the wording like that, either of those phrases will be clear enough by themselves to someone who can't imagine why/when the OS would do this.  It might be worth adding an example like "the job launcher (srun/aprun) has a flag telling the OS how many cores/PUs/threads to make available to the process" (if that is indeed one).